src/snowflake/snowpark/mock/_analyzer.py

```diff
@@ -511,60 +611,91 @@ def table_function_expression_extractor(
     def unary_expression_extractor(
         self,
         expr: UnaryExpression,
-        expr_to_alias: Dict[str, str],
+        df_aliased_col_name_to_real_col_name: DefaultDict[str, Dict[str, str]],
         parse_local_name=False,
         keep_alias=True,
     ) -> str:
         if isinstance(expr, Alias):
             quoted_name = quote_name(expr.name)
             if isinstance(expr.child, Attribute):
-                expr_to_alias[expr.child.expr_id] = quoted_name
-                for k, v in expr_to_alias.items():
+                self.generated_alias_maps[expr.child.expr_id] = quoted_name
+                assert self.alias_maps_to_use is not None
+                for k, v in self.alias_maps_to_use.items():
                     if v == expr.child.name:
-                        expr_to_alias[k] = quoted_name
+                        self.generated_alias_maps[k] = quoted_name
+
+                if df_aliased_col_name_to_real_col_name:
+                    for df_alias_dict in df_aliased_col_name_to_real_col_name.values():
+                        for k, v in df_alias_dict.items():
+                            if v == expr.child.name:
+                                df_alias_dict[k] = quoted_name
+
             alias_exp = alias_expression(
-                self.analyze(expr.child, expr_to_alias, parse_local_name), quoted_name
+                self.analyze(
+                    expr.child, df_aliased_col_name_to_real_col_name, parse_local_name
+                ),
+                quoted_name,
             )
 
             expr_str = alias_exp if keep_alias else expr.name or keep_alias
             expr_str = expr_str.upper() if parse_local_name else expr_str
             return expr_str
         if isinstance(expr, UnresolvedAlias):
-            expr_str = self.analyze(expr.child, expr_to_alias, parse_local_name)
-            if parse_local_name:
-                expr_str = expr_str.upper()
-            return quote_name(expr_str.strip())
+            expr_str = self.analyze(
+                expr.child, df_aliased_col_name_to_real_col_name, parse_local_name
+            )
+            if isinstance(expr_str, str):
+                if parse_local_name:
+                    expr_str = expr_str.upper()
+                return quote_name(expr_str.strip())
+            elif isinstance(expr_str, list):
+                assert all(isinstance(e, str) for e in expr_str)
+                return ",".join([quote_name(e.strip()) for e in expr_str])
+            else:
+                raise SnowparkLocalTestingException(
+                    f"Invalid expression '{expr_str}' for UnresolvedAlias."
+                    f" Please ensure the alias name exists in the DataFrame."
+                )
```

Comment:
Should there be an added test case for this branch?

Reply:
this case should not happen.
I changed to assertion at the beginning and removed this error